test/lit/passes/cast-refining.wast

```diff
@@ -0,0 +1,1124 @@
+;; NOTE: Assertions have been generated by update_lit_checks.py --all-items and should not be edited.
+
+;; RUN: foreach %s %t wasm-opt --cast-refining --traps-never-happen -all --closed-world --nominal -S -o - | filecheck %s --check-prefix=YESTNH
+;; RUN: foreach %s %t wasm-opt --cast-refining                      -all --closed-world --nominal -S -o - | filecheck %s --check-prefix=NO_TNH
```

Comment:
How about running these tests with `--hybrid` instead of `--nominal` in the hopes that we can eventually remove nominal typing? You'll just want to add a `--remove-unused-types` at the end so the unused types actually get removed from the test output.

Reply:
Do we want to remove nominal? I think it's useful in the long term too, and it makes tests easier to read and reason about, at least for me.

Reply:
Ideally I would like to make `--nominal` into a parser flag at most and transition away from using it ourselves so our test files can be fully valid Wasm. The only thing stopping me from doing that now is that our closed-world validation means that we would have to be more sophisticated than just sticking all the types in the same rec group.

Reply:
I see. Sounds worth discussing.

Needing to add `--remove-unused-types` here though seems like a downside to me. These tests are a good example of where nominal typing makes things simpler.